src/Select/Select.test.tsx

```diff
@@ -1,13 +1,20 @@
-import React from 'react'
+import React, { forwardRef } from 'react'
+
 import { mountWithTheme } from '../utils/tests/mountWithTheme'
 
 import { Select } from '.'
-import { injectComponent } from './Select'
-import { SelectInput } from '../SelectInput'
+
+const CustomInput = forwardRef<any, any>((props, ref) => (
+  <input id='test-id' {...props} ref={ref} />
+))
 
 test('SelectT injects value component with selected value', () => {
   const wrapper = mountWithTheme((
-    <Select value={{ key: 'one', value: 'one' }} data={[]} />
+    <Select
+      value={{ key: 'one', value: 'one' }}
```

Comment:
So i'm not too sure about this. A regular select takes a string as value (although that's also the case with options which we cant do because we need unique identifiers).  However it seems more logical to me to have a string primitive as value as its most likely that a value will be stored in state in that format somewhere.  That would mean we need to call additional props in most places. Have you considered following option to perhaps turn it around?
```
 <Select
      // default as string
      value={state || undefined}
      // unless valueAccessor
      valueAccessor={(item) => item.key}
      useInputComponentWidth={false}
      placeholder='Choose option'
      onChange={storeState}
      data={items}
    />
```

Reply:
I like the possibility to use this in multiple ways. The last one (with passing in own Option components) is probably not very feasible with react-window though.

I agree the default should be:

```tsx
<Select value={T | undefined} data={T[]} onChange={(T) => void} />
```

Which could expand internally to the custom implementation:

```tsx
<Select value={T | undefined}>
  <Options selected={T | undefined} data={T[]} onChange={(T) => void}/>
</Select>
```

Reply:
Default:
```JSX
<Select value='string value'>
 <SelectList
    data={some_data_here}
    onChange={hande_changes}
    value='the same string, as the one for <Select> component'
  />
</Select>
```

Custom input:
```JSX
<Select
  value='string value'
  InputComponent={({ value, placeholder, disabled }) => CustomInput}
>
 <SelectList
    data={some_data_here}
    onChange={hande_changes}
    value='the same string, as the one for <Select> component'
  />
</Select>
```
What you think?

Reply:
Let's go with this implementation and revise later if there's a need for a slightly different way to handle data (e.g. support array of strings out of the box, etc.). Let's move this forward, so I'm resolving this conversation. (We can always review changes again)